### PR TITLE
Update to UCX 1.11.0

### PR DESCRIPTION
**Pull request by abellina**
Signed-off-by: Alessandro Bellina <abellina@nvidia.com>

Updates docs and pom to use the UCX 1.11 jar and point to the correct artifacts from https://github.com/openucx/ucx/releases/tag/v1.11.0.

I checked the LICENSE file for 1.11.0, and it did not change from our prior snapshot.

Closes: https://github.com/NVIDIA/spark-rapids/issues/2925

### CODE REVIEW

**Comment by abellina** [comment]
I forgot to add a note on enhanced compatibility. I'll add something shortly.

File: docs/additional-functionality/rapids-shuffle.md [hunk 1]
```diff
@@ -65,52 +65,48 @@ The minimum UCX requirement for the RAPIDS Shuffle Manager is
...
-   RDMA packages have extra requirements that should be satisfied by MLNX_OFED.
-  
-   ---
-   **NOTE:**
+2. Fetch and install the UCX package for your OS from:
+   [UCX 1.11.0](https://github.com/openucx/ucx/releases/tag/v1.11.0).
+
+   NOTE: Please install the artifact with the newest CUDA 11.x version (for UCX 1.11.0 please
+   pick CUDA 11.2) as CUDA 11 introduced [CUDA Enhanced Compatibility](https://docs.nvidia.com/deploy/cuda-compatibility/index.html#enhanced-compat-minor-releases). 
+   Starting UCX 1.12, UCX will stop publishing individual artifacts for each minor version of CUDA.
```
Comment (jlowe) [inline docs/additional-functionality/rapids-shuffle.md:73]:
```suggestion
   Starting with UCX 1.12, UCX will stop publishing individual artifacts for each minor version of CUDA.
```

**Review by jlowe — APPROVED** [review]
Minor comment, lgtm.

**Comment by abellina** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**